### PR TITLE
Allow more WhatsApp templates to be usable in the flows

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/utils/whatsapp/tasks.py [hunk 1]
```diff
@@ -159,23 +159,17 @@ def update_local_templates(channel, templates_data):
         params, transformed_components = _extract_template_params(components)
         content_parts = []
 
-        all_supported = True
         for component in components:
-            if component["type"] not in ["HEADER", "BODY", "FOOTER"]:
-                continue
+            if component["type"] in ["HEADER", "BODY", "FOOTER"]:
```
Comment (rowanseymour) [inline temba/utils/whatsapp/tasks.py:163]: but why are we still limiting to these ?

Comment (norkans7) [reply temba/utils/whatsapp/tasks.py:163]: We need to make sure we have text in at least one of those components

I think we still do not have support for templates such as the OTP kind that expire in time 
We have one in the tests 

https://github.com/nyaruka/rapidpro/pull/5095/files#diff-766c88132c33459d25a1214a036b91efbe9f5ce2239ba46c243cdb9d795e552cL449-L462

Comment (norkans7) [reply temba/utils/whatsapp/tasks.py:163]: I think we accept those with a text body in any of the components

Comment (rowanseymour) [reply temba/utils/whatsapp/tasks.py:163]: Ok then let's make the check be for text instead of type ?

Comment (norkans7) [reply temba/utils/whatsapp/tasks.py:163]: the logic here is that we check we have text if we have on of those type we do not want to check the text only as the buttons do not have text either and are still valid components we support

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**